### PR TITLE
move cors into common config

### PR DESCRIPTION
**Pull request by allencloud**
1.move cors into common config
2.so windows config can use cors, too

### CODE REVIEW

**Comment by calavera** [comment]
LGTM

**Comment by jhowardmsft** [comment]
+1 to what @calavera said - the config flags need moving back to in `config-unix.go`, but not the deprecated one. That should remain linux specific as there's no reason to introduce something on Windows that's going to be deprecated. There's also setPlatformServerConfig in `daemon_unix.go` which needs changing.

File: daemon/config.go [hunk 1]
```diff
@@ -128,6 +130,8 @@ func (config *Config) InstallCommonFlags(cmd *flag.FlagSet, usageFn func(string)
 	cmd.StringVar(&config.ClusterAdvertise, []string{"-cluster-advertise"}, "", usageFn("Address or interface name to advertise"))
 	cmd.StringVar(&config.ClusterStore, []string{"-cluster-store"}, "", usageFn("Set the cluster store"))
 	cmd.Var(opts.NewNamedMapOpts("cluster-store-opts", config.ClusterOpts, nil), []string{"-cluster-store-opt"}, usageFn("Set cluster store options"))
+	cmd.BoolVar(&config.EnableCors, []string{"#api-enable-cors", "#-api-enable-cors"}, false, usageFn("Enable CORS headers in the remote API, this is deprecated by --api-cors-header"))
```
Comment (calavera) [inline daemon/config.go]: this flag  is deprecated, we don't want to move it.

File: daemon/config.go [hunk 2]
```diff
@@ -70,6 +70,8 @@ type CommonConfig struct {
 	Root                 string              `json:"graph,omitempty"`
 	SocketGroup          string              `json:"group,omitempty"`
 	TrustKeyPath         string              `json:"-"`
+	CorsHeaders          string              `json:"api-cors-headers,omitempty"`
+	EnableCors           bool                `json:"api-enable-cors,omitempty"`
```
Comment (calavera) [inline daemon/config.go]: This field can also stay in were it was before, we only need `CorsHeaders` in both platforms.

Comment (calavera) [inline daemon/config.go]: Oh, nevermind, I just realized that we need both fields.

**Comment by calavera** [comment]
LGTM

**Comment by jhowardmsft** [comment]
Still needs the other changes I mentioned ^^
Does not LGTM

**Comment by calavera** [comment]
> There's also setPlatformServerConfig in daemon_unix.go which needs changing.

oh, right, I missed this. @allencloud can you take a look at that function?

**Comment by allencloud** [comment]
sorry for my delay. @calavera @jhowardmsft 
I have changed that function. however it seems there was a test `gccgo` which went wrong.

File: docker/daemon_unix.go [hunk 3]
```diff
@@ -23,9 +23,6 @@ import (
 const defaultDaemonConfigFile = "/etc/docker/daemon.json"
 
 func setPlatformServerConfig(serverConfig *apiserver.Config, daemonCfg *daemon.Config) *apiserver.Config {
```
Comment (calavera) [inline docker/daemon_unix.go]: I think at this point we can remove this function and it's call in `daemon/daemon.go` L222. We can also remove the windows version in `daemon/daemon_windows.go`.

Comment (jhowardmsft) [inline docker/daemon_unix.go]: Yup, good point.

**Comment by allencloud** [comment]
ping @calavera @jhowardmsft

**Comment by vdemeester** [comment]
LGTM 🐶

**Comment by calavera** [comment]
LGTM

**Pull request merged**